### PR TITLE
Extract fees from Treasury

### LINKED ISSUE #3002: Extract fees from Treasury for reward distribution
## What is the Problem Being Solved?

The contents of the Treasury's reward pool should be extracted at regular intervals and provided to something (see #3001) that will distribute rewards to stakers.

## Description of the Design

creator facet of the treasury module will have makeMakeCollectFeesInvitation (like https://github.com/Agoric/agoric-sdk/pull/2975/files). There will be a separate contract that holds that facet, and has a timer that signals the end of the epoch. Each time the timer goes off, it'll feed accumulated fees to another contract

### PR DESCRIPTION
**Pull request by Chris-Hibbert**
includes both system fees to the treasury, and protocol fees to the AMM.

closes #3002

### CODE REVIEW

File: packages/treasury/src/collectRewardFees.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import { offerTo } from '@agoric/zoe/src/contractSupport';
+import { E } from '@agoric/eventual-send';
+
+export const makeMakeCollectFeesInvitation = (
+  zcf,
+  feeSeat,
+  autoswapCreatorFacet,
+  runBrand,
+) => {
+  const collectFees = async seat => {
```
Comment (dtribble) [inline packages/treasury/src/collectRewardFees.js:13]: Are we no longer using a naming pattern for hooks?

File: packages/treasury/src/collectRewardFees.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    const invitation = await E(
```
Comment (dtribble) [inline packages/treasury/src/collectRewardFees.js:14]: Does this really need to wrap in our coding convention?

File: packages/treasury/src/collectRewardFees.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    ).makeCollectFeesInvitation();
+    const { zcfSeat: transferSeat } = zcf.makeEmptySeatKit();
+    await E.get(offerTo(zcf, invitation, {}, {}, transferSeat)).deposited;
+
+    const totalTransferred = amountMath.add(
+      feeSeat.getAmountAllocated('RUN', runBrand),
+      transferSeat.getAmountAllocated('RUN', runBrand),
+    );
+    zcf.reallocate(
+      transferSeat.stage({ RUN: amountMath.makeEmpty(runBrand) }),
```
Comment (dtribble) [inline packages/treasury/src/collectRewardFees.js]: make the empty into a shared const?

File: packages/treasury/src/stablecoinMachine.js [hunk 4]
```diff
@@ -92,13 +93,20 @@ export async function start(zcf) {
...
+    publicFacet: autoswapAPI,
+    instance: autoswapInstance,
+    creatorFacet: autoswapCreatorFacet,
+  } = await E(zoe).startInstance(
     autoswapInstall,
     { Central: runIssuer },
-    { timer: timerService },
+    {
+      timer: timerService,
+      poolFee: loanParams.poolFee,
```
Comment (dtribble) [inline packages/treasury/src/stablecoinMachine.js:108]: The treasury should not be so informed about the structure of the AMM. We will need to refactor the creation so they are instantiated more independently.

File: packages/treasury/test/test-stablecoin.js [hunk 5]
```diff
@@ -1355,6 +1369,18 @@ test('overdeposit', async t => {
     aliceCollateralization5.denominator,
     amountMath.make(1n, runBrand),
   );
+
+  const collectFeesSeat = await E(zoe).offer(
+    E(stablecoinMachine).makeCollectFeesInvitation(),
+    {},
```
Comment (dtribble) [inline packages/treasury/test/test-stablecoin.js]: These arguments are optional. 
```
  const offer = async (
    invitation,
    uncleanProposal = harden({}),
    paymentKeywordRecord = harden({}),
  ```

File: packages/treasury/test/test-stablecoin.js [hunk 6]
```diff
@@ -1355,6 +1369,18 @@ test('overdeposit', async t => {
...
+    {},
+  );
+  await E(collectFeesSeat).getOfferResult();
+  assertAmountsEqual(
+    t,
+    await E.get(E(collectFeesSeat).getCurrentAllocation()).RUN,
```
Comment (dtribble) [inline packages/treasury/test/test-stablecoin.js:1378]: `await` expression not allowed (though tests may be an exception)

**Review by dtribble — APPROVED** [review]
Some optional feedback and future topics

File: packages/treasury/src/stablecoinMachine.js (hunk 4 above)
Comment (Chris-Hibbert) [reply packages/treasury/src/stablecoinMachine.js:108]: added #3019

**Pull request merged**